### DISCUSSION TITLE
用 h 函数或者 createVnde 去创建一个 dom 和直接用 jsx 写一个相比，这种有什么优点

### DISCUSSION THREAD

**Discussion by Miofly** [element-plus/element-plus#10227] [Q&A] [answered]
在什么场景下组件用 h 函数 或者 jsx 或者 template，现在对这个比较迷惑，感觉组件库中的代码，这几种形式都有用到，用 jsx 替代 h 函数，这种在性能上有什么影响吗

![image](https://user-images.githubusercontent.com/42859910/197695967-19adb9b5-ab6c-4131-93e1-0576ece12888.png)

**Comment by webvs2** [comment, marked as answer]
vue是运行和编译时框架，用了jsx去写，也需要再编译成Vnode，和H函数出来的一样。但是用H函数出来就免去编译了，只是jsx更好的阅读而已。
